lib/jekyll/assets/liquid/tag/proxied_asset.rb

```diff
@@ -94,13 +107,30 @@ def digest_path
             "#{name.chomp(ext)}-#{digest}#{ext}"
           end
 
+          # --------------------------------------------------------------------
+
+          private
+          def _mime_for(ext)
+            Sprockets.mime_exts[ext]
+          end
+
+          # --------------------------------------------------------------------
+
+          private
+          def _ext_for(type)
+            Sprockets.mime_types[type][:extensions].first
+          end
+
           # ------------------------------------------------------------------
 
           private
           def proxy_file
             unless cached?
               args.proxies.each do |key, val|
+                old_type = content_type
+                old_filename = filename
                 Proxies.get(key).first[:class].new(self, val, @args).process
+                File.delete old_filename if old_type != content_type
```

Comment:
This should use `Pathutil` not file. and filename should already be an instance of Pathutil.